src/app/clusters/pump-configuration-and-control-server/pump-configuration-and-control-server.cpp

```diff
@@ -450,11 +397,11 @@ void MatterPumpConfigurationAndControlClusterServerAttributeChangedCallback(cons
     switch (attributePath.mAttributeId)
     {
     case Attributes::ControlMode::Id: {
-        updateAttributeLinks(attributePath.mEndpointId);
+        setEffectiveOperationAndControlMode(attributePath.mEndpointId);
     }
     break;
     case Attributes::OperationMode::Id: {
-        updateAttributeLinks(attributePath.mEndpointId);
+        setEffectiveOperationAndControlMode(attributePath.mEndpointId);
     }
```

Comment:
case Attributes::ControlMode::Id:
case Attributes::OperationMode::Id:
      setEffectiveMode(attributePath.mEndpointId);
break;

Reply:
Good catch, I will update this

Reply:
Is fixed in the latest commit